From 7ee10734d084ba84c0f72f49ce563ea5058e12a4 Mon Sep 17 00:00:00 2001
From: Prasad Mujumdar <prasadm@cloudera.com>
Date: Mon, 12 May 2014 18:32:39 -0700
Subject: [PATCH 0023/1164] CLOUDERA-BUILD:  CDH-19069:Hive script should
 include jar from SENTRY_HOME into classpath

(cherry picked from commit 1a65cf6fce3818876cbfef9e1c6a57d5cadf56c0)
(cherry picked from commit 46f8e7d72f4b7ace1f81ec081017c650ebfade30)

Conflicts:
	bin/ext/hiveserver2.sh
---
 bin/ext/hiveserver2.sh |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/bin/ext/hiveserver2.sh b/bin/ext/hiveserver2.sh
index 42d3d79..39e3d39 100644
--- a/bin/ext/hiveserver2.sh
+++ b/bin/ext/hiveserver2.sh
@@ -23,6 +23,21 @@ hiveserver2() {
   fi
   JAR=${HIVE_LIB}/hive-service-*.jar
 
+  # Set SENTRY_HOME if possible and add Sentry jars to classpath
+  if [[ -z "$SENTRY_HOME" ]]
+  then
+    if [[ -d ${HIVE_HOME}/../sentry ]]
+    then
+      export SENTRY_HOME=`readlink -m ${HIVE_HOME}/../sentry`
+    fi
+  fi
+  if [[ -n "$SENTRY_HOME" ]]
+  then
+    for f in ${SENTRY_HOME}/lib/*.jar; do
+      export HADOOP_CLASSPATH=${HADOOP_CLASSPATH}:${f}
+    done
+  fi
+
   exec $HADOOP jar $JAR $CLASS $HIVE_OPTS "$@"
 }
 
-- 
1.7.9.5

